src/components/engagement/engagement.repository.ts

```diff
@@ -71,32 +71,33 @@ export class EngagementRepository extends CommonRepository {
           .match([
             node('project'),
             relation('out', '', 'engagement', {
-              active: view?.type !== 'deleted',
+              active: !view?.deleted,
             }),
             node('node', label, { id }),
           ])
           .return('project, node')
           .apply((q) =>
-            changeset
+            view?.changeset
               ? q
                   .union()
                   .match([
                     node('project'),
                     relation('out', '', 'engagement', { active: false }),
                     node('node', label, { id }),
                     relation('in', '', 'changeset', { active: true }),
-                    node('changeset', 'Changeset', { id: changeset }),
+                    node('changeset', 'Changeset', { id: view.changeset }),
                   ])
                   .return('project, node')
               : q
           )
       )
       .apply(
-        matchPropsAndProjectSensAndScopedRoles(session, {
-          deleted: view?.type === 'deleted',
-        })
+        matchPropsAndProjectSensAndScopedRoles(
+          session,
+          view?.deleted ? { view } : {}
```

Comment:
We should pass the `view` only for `deleted`... I am not sure what is the best option here.

Reply:
I was suggesting always pass it. It could still be nullable though.